autoload/go/impl.vim

```diff
@@ -101,13 +101,29 @@ function! s:root_dirs() abort
   return dirs
 endfunction
 
-function! s:go_packages(dirs) abort
+function! s:go_packages(dirs, arglead) abort
   let pkgs = []
-  for d in a:dirs
-    let pkg_root = expand(d . '/pkg/' . go#util#osarch())
-    call extend(pkgs, split(globpath(pkg_root, '**/*.a', 1), "\n"))
+  for dir in a:dirs
+      " this may expand to multiple lines
+      let scr_root = expand(dir . '/src/')
+      for pkg in split(globpath(scr_root, a:arglead.'*'), "\n")
+          if isdirectory(pkg)
+              let pkg .= '/'
+          elseif pkg !~ '\.a$'
```

Comment:
This part was straight copied from `go#path#Complete` which we use for `:GoImport`. I didn't investigated it to be honest :)

Reply:
Ah, I thought that might be the case. The code in `go#package#Complete` looks in `$GOPATH/pkg`, where there may be `*.a` files. Since this isn't looking in `$GOPATH/pkg`, this could be a simple `else`.